api/src/main/resources/workbench.yaml

```diff
@@ -1050,7 +1050,12 @@ paths:
         - workspaces
       consumes:
         - application/json
-      description: Shares a workspace with users
+      description: >
+        Shares a workspace with the given users and permissions. Note that the entire ACL must
```

Comment:
This seems non intuitive to me and doesn't really reflect the intent of the user IMO. "I want to add a user to the share list" vs "I want to set the share list to this + someone new".

Not suggesting we change this now but do you think it would be worth changing in the future? Aside from the benefit of (IMO)  a simpler API, it looks like we'll gain a functional improvement in decreased loading times by not requiring the fetch ACL call.

Reply:
I mostly agree!

The challenge is that this API and the share-workspace UI component are both used in two places: here in the workspace-list page, but also in the workspace about page. In the context of the workspace list, it makes sense to be a "quickly add one collaborator" type action. But from the about page, it's explicitly about editing / removing the ACLs across the whole workspace.

If we split out a new API, then we'll be maintaining two highly-related API endpoints instead of one. It may be a net gain, but is certainly beyond the scope of this bugfix.

I'm not sure what the right motivation might be to follow-up on a decision here... it's not enough of an obvious win for me to want to immediately track a tech debt ticket, but I could be convinced otherwise.

(I think we'll bump up against this API again when we start allowing users to self-publish workspaces, since it will involve making sure these share actions are round-trip safe even when we have a "all users" entry in the Terra ACL. At that point we may want to make our APIs less of a pass-through to Firecloud.